surf.bundle.js

```diff
@@ -0,0 +1,6956 @@
+                },
+                settingsHamburgerMenu_flyout_label: {
+                    margin: "0 0 0.5em",
+                    display: "block"
+                },
+                settingsHamburgerMenu_flyout_score: {
+                    marginTop: "8px",
+                    display: "flex",
+                    alignItems: "center"
+                },
+                settingsHamburgerMenu_flyout_scoreCrown: {
+                    height: "16px",
+                    width: "16px",
+                    color: "currentColor",
+                    fill: "currentColor",
+                    marginInlineEnd: "4px"
+                },
+                settingsHamburgerMenu_flyout_toggleRow: {
+                    display: "flex",
+                    flexDirection: "row",
+                    justifyContent: "space-between",
+                    padding: "6px 0"
+                },
+                settingsHamburgerMenu_centered: {
+                    margin: "0 auto"
+                },
+                settingsHamburgerMenu_icon: {
+                    color: d.c,
+                    fill: "currentColor"
+                },
+                settingsHamburgerMenu_creditsTable: {
+                    margin: "0 auto",
+                    color: d.c
+                },
+                settingsHamburgerMenu_creditsTable_specialThanks: {
+                    marginTop: "8px",
+                    marginBottom: "4px"
+                }
+            };
+        var w, f = t(149),
+            S = t(619),
+            k = t(289),
+            O = t(144),
+            x = t(200),
+            v = t(633),
+            C = t(109),
+            E = t(431),
+            T = t(644),
+            P = t(630),
+            M = t(613),
+            I = t(618),
+            A = t(634),
+            D = t(2),
+            L = t(91),
+            B = t(454),
+            R = t(41),
+            j = t(611),
+            N = t(287),
+            G = t(1);
+        ! function(e) {
+            e.Buoy = "BUOY", e.Classic = "CLASSIC", e.Speed = "SPEED"
+        }(w || (w = {}));
+        class W {
+            static recordNewGame(e) {
+                switch (e) {
+                    case w.Buoy:
+                        W.numBuoyGames += 1;
+                        break;
+                    case w.Classic:
+                        W.numClassicGames += 1;
+                        break;
+                    case w.Speed:
+                        W.numSpeedGames += 1
+                }
+            }
+            static defaultStats() {
+                return {
+                    buoy_bestScore: -1,
+                    classic_bestScore: -1,
+                    currentCharacter: 4,
+                    highVisibilityMode: false,
+                    mode: "CLASSIC",
+                    reducedSpeedMode: false,
+                    speed_bestScore: -0.001
+                }
+            }
+            static getAllStats() {
```

Comment:
This routine has a problem in that the internal "stats" may be invalid and if it is, the game won't load. This is called `onload` from surf.bundle.js@6714. There should probably be another function in this class which validates the result of the `localStorage` call and if the result is invalid it rebuilds the default stats. This would also resolve resetAllStats making it so that reseting the stats is just deleting the storage... probably a better fix anyway. I may check in a quick fix for just that anyway, but the validation routine should be addressed later if someone wants to tackle it.